src/persist-types/src/schema.rs

```diff
@@ -88,16 +89,35 @@ impl ArrayMigration {
             NoOp => array,
             Struct(migrations) => {
                 let len = array.len();
-                let array = array
-                    .as_any()
-                    .downcast_ref::<StructArray>()
-                    .unwrap_or_else(|| panic!("expected Struct got {:?}", array.data_type()))
-                    .clone();
-                let (mut fields, mut arrays, nulls) = array.into_parts();
-                for migration in migrations {
-                    migration.migrate(len, &mut fields, &mut arrays);
+
+                match array.data_type() {
```

Comment:
nit: I'd structure this something like the following to avoid the duplication

```
let (mut fields, mut arrays, nulls) = match array.data_type() {
    DataType::Null => (Fields::empty(), Vec::new(), None),
    DataType::Struct(_) => {
                        let array = array
                            .as_any()
                            .downcast_ref::<StructArray>()
                            .expect("known to be StructArray")
                            .clone();
                        array.into_parts()
    },
};
```

Reply:
Fixed!